tests/test_secondary.py

```diff
@@ -0,0 +1,721 @@
+"""
+<Program Name>
+  test_secondary.py
+
+<Purpose>
+  Unit testing for uptane/clients/secondary.py
+  Much of this is copied from test_primary.py and then modified slightly
+"""
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import uptane # Import before TUF modules; may change tuf.conf values.
+
+import unittest
+import os.path
+import time
+import shutil
+import hashlib
+
+from six.moves.urllib.error import URLError
+
+import tuf
+import tuf.formats
+import tuf.conf
+import tuf.client.updater
+
+import uptane.formats
+import uptane.clients.secondary as secondary
+import uptane.common # verify sigs, create client dir structure, convert key
+import uptane.encoding.asn1_codec as asn1_codec
+
+# For temporary convenience:
+import demo # for generate_key, import_public_key, import_private_key
+
+TEST_DATA_DIR = os.path.join(uptane.WORKING_DIR, 'tests', 'test_data')
```

Comment:
I haven't considered it yet. This works and is what is already done in test_primary. Should consider it later.